clang/lib/Sema/SemaDeclAttr.cpp

```diff
@@ -3523,6 +3523,23 @@ static bool InvalidWorkGroupSizeAttrs(const Expr *MGValue, const Expr *XDim,
            ZDimExpr->getResultAsAPSInt() != 1));
 }
 
+// If the [[intel::max_work_group_size(X, Y, Z)]] attribute is specified on
+// a declaration along with [[sycl::reqd_work_group_size(X1, Y1, Z1)]]
+// attribute, check to see if values of reqd_work_group_size arguments are
+// equal or less than values of max_work_group_size attribute arguments.
+static bool checkWorkGroupSizeAttrValues(const Expr *LHS, const Expr *RHS) {
+  // If any of the operand is still value dependent, we can't test anything.
+  const auto *LHSCE = dyn_cast<ConstantExpr>(LHS);
+  const auto *RHSCE = dyn_cast<ConstantExpr>(RHS);
+
+  if (!LHSCE || !RHSCE)
+    return false;
+
+  // Otherwise, check if value of reqd_work_group_size argument is
+  // equal or less than value of max_work_group_size attribute argument.
+  return !(LHSCE->getResultAsAPSInt() <= RHSCE->getResultAsAPSInt());
```

Comment:
Isn't this just `>`?

Reply:
Thanks @AaronBallman for the reviews.

According to https://github.com/intel/llvm/blob/68c459a7fc3c24d867a63deb0f45804d746cb815/clang/lib/Sema/SemaDeclAttr.cpp#L3238-#L3241

// Checks correctness of mutual usage of different work_group_size attributes:
// reqd_work_group_size, max_work_group_size and max_global_work_dim.
// Values of reqd_work_group_size arguments shall be equal or less than values
// coming from max_work_group_size.